File: D_Destruction_of_the_Dandelion_Fields.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long
#define yes cout<<"YES"<<endl
#define no cout<<"NO"<<endl
#define fast_io ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
int main()
{
    fast_io;

    int t;
    cin>>t;
    while(t--)
    {
        int n;
        cin>>n;
        vector<int>v(n);
        for(int i=0;i<n;i++)
        {
            cin>>v[i];
        }
        
        vector<int>even,odd;
        for(int i=0;i<n;i++)
        {
            if(v[i]&1)odd.push_back(v[i]);
            else even.push_back(v[i]);
        }
        sort(even.rbegin(),even.rend());
        sort(odd.rbegin(),odd.rend());

        if(odd.size()==0)
        {
            cout<<0<<endl;
            continue;
        }

        ll esum=0;
        for(auto u:even)
        {
            esum+=u;
        }
        int indx=1;
        if(odd.size()>1)
        indx= (odd.size()+1)/2;
        else indx=1;
        for(int i=0;i<indx;i++)
        {
            esum+=odd[i];
        }
        cout<<esum<<endl;
    }
    return 0;
}

File: D_Destruction_of_the_Dandelion_Fields.py
t=int(input())
for _ in range(t):
    n=int(input())
    li=list(map(int,input().split()))
    even=[]
    odd=[]

    for num in li:
        if(num&1):odd.append(num)
        else: even.append(num)
    

    even.sort(reverse=True)
    odd.sort(reverse=True)

    if(len(odd)==0):
        print(0)
        continue

    ans=0
    for val in even:
        ans+=val

    
    indx=(len(odd)+1)//2

    for i in range(0,indx):
        ans+=odd[i]

    print(ans)